### PR TITLE
fix(release): publish to npm workflow fails when set project package manager to PNPM

### PR DESCRIPTION
**Pull request by huaxk**
Fixes: #2078

Add pnpm/action-setup step into jsii-project prePublishSteps.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by mrgrain — CHANGES_REQUESTED** [review]
Thanks for this fix @huaxk ! Would you mind adding a test case for this scenario?

File: src/cdk/jsii-project.ts [hunk 1]
```diff
@@ -432,27 +432,37 @@ export class JsiiProject extends TypeScriptProject {
...
-        },
-        {
-          name: "Install Dependencies",
-          run: `cd ${REPO_TEMP_DIRECTORY} && ${this.package.installCommand}`,
-        },
-        {
-          name: `Create ${target} artifact`,
-          run: `cd ${REPO_TEMP_DIRECTORY} && npx projen ${packTask.name}`,
-        },
+    let prePublishSteps: Array<Step> = [
```
[the remark this answers is not in the document]
Comment (huaxk) [reply src/cdk/jsii-project.ts:435]: OK!

File: test/cdk/jsii.test.ts [hunk 2]
```diff
@@ -566,3 +567,46 @@ describe("workflows use global workflowRunsOn option", () => {
...
+    publishToNuget: {
+      dotNetNamespace: "DotNet.Namespace",
+      packageId: "PackageId",
+    },
+    publishToPypi: { distName: "dist-name", module: "module-name" },
+    packageManager: javascript.NodePackageManager.PNPM,
+  });
+
+  const output = synthSnapshot(project);
+  // const build = yaml.parse(output[".github/workflows/build.yml"]);
```
Comment (mrgrain) [inline test/cdk/jsii.test.ts:593]: Any reason this was left in?
```suggestion
```

File: test/cdk/jsii.test.ts [hunk 3]
```diff
@@ -566,3 +567,46 @@ describe("workflows use global workflowRunsOn option", () => {
...
+  const release = yaml.parse(output[".github/workflows/release.yml"]);
+  // const tasks = output[".projen/tasks.json"].tasks;
```
Comment (mrgrain) [inline test/cdk/jsii.test.ts:595]: Any reason this was left in?
```suggestion
```

**Review by mrgrain — CHANGES_REQUESTED** [review]
This looks great now! Just these two minor comments and we are good to go. 😃

**Review by mrgrain — APPROVED** [review]
🥳  Thanks again for providing this fix!

**Pull request merged**